src/shortcut.js

```diff
@@ -0,0 +1,330 @@
+  Blockly.ShortcutRegistry.registry.addKeyMapping(metaX, cutShortcut.name);
+};
+
+/**
+ * Keyboard shortcut to paste multiple selected blocks on
+ * ctrl+v, cmd+v, or alt+v.
+ */
+const registerPaste = function() {
+  const pasteShortcut = {
+    name: Blockly.ShortcutItems.names.PASTE,
+    preconditionFn: function(workspace) {
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress();
+    },
+    callback: function() {
+      Blockly.Events.setGroup(true);
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        if (block) {
+          block.pathObject.updateSelected(false);
+        }
+      });
+      blockSelection.clear();
+      copyData.forEach(function(data) {
+        if (!data) {
+          return null;
+        }
+        // Pasting always pastes to the main workspace, even if the copy
+        // started in a flyout workspace.
+        let workspace = data.source;
+        if (workspace.isFlyout) {
+          workspace = workspace.targetWorkspace;
+        }
+        if (data.typeCounts &&
+            workspace.isCapacityAvailable(data.typeCounts)) {
+          const block = workspace.paste(data.saveInfo);
+          block.pathObject.updateSelected(true);
+          blockSelection.add(block.id);
+          return block;
+        }
+        return null;
+      });
+      Blockly.Events.setGroup(false);
+      return true;
+    },
+  };
+
+  Blockly.ShortcutRegistry.registry.register(pasteShortcut);
+
+  const ctrlV = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.V, [Blockly.utils.KeyCodes.CTRL]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(ctrlV, pasteShortcut.name);
+
+  const altV =
+  Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.V, [Blockly.utils.KeyCodes.ALT]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(altV, pasteShortcut.name);
+
+  const metaV = Blockly.ShortcutRegistry.registry.createSerializedKey(
+      Blockly.utils.KeyCodes.V, [Blockly.utils.KeyCodes.META]);
+  Blockly.ShortcutRegistry.registry.addKeyMapping(metaV, pasteShortcut.name);
+};
+
+/**
+ * Keyboard shortcut to select all top blocks in the workspace on
+ * ctrl+a, cmd+a, or alt+a.
+ */
+const registeSelectAll = function() {
+  const selectAllShortcut = {
+    name: 'selectall',
+    preconditionFn: function(workspace) {
+      return !workspace.options.readOnly && !Blockly.Gesture.inProgress();
+    },
+    callback: function(workspace, e) {
+      // Prevent the default text all selection behavior.
+      e.preventDefault();
+      if (Blockly.selected) {
+        Blockly.selected.pathObject.updateSelected(false);
+        Blockly.common.setSelected(null);
+      }
+      workspace.getTopBlocks().forEach(function(block) {
+        if (block &&
+            block.isDeletable() &&
+            block.isMovable() &&
+            !block.isInsertionMarker()) {
```

Comment:
```suggestion
        if (block &&
            (block.isDeletable() || block.isMovable()) &&
            !block.isInsertionMarker()) {
```

I think the block should probably be selected as long as it's either deletable or movable. Because in that case there are some group operations that could apply to it.